spec/models/embedded_ansible_worker/runner_spec.rb

```diff
@@ -10,7 +10,9 @@
     let(:runner) {
       worker
       allow_any_instance_of(described_class).to receive(:worker_initialization)
-      described_class.new(:guid => worker_guid)
+      r = described_class.new(:guid => worker_guid)
+      allow(r).to receive(:worker).and_return(worker)
```

Comment:
So I tried to do a bit to get rid of this and it turns out we have to jump through hoops because `sync_config` changes the server's zone to the one called "default" which ends up being `nil` unless we `Zone.seed` ...

Even then there is no way to deal with how the runner looks up the worker, I would still need to stub the `.worker` method (attribute) to make sure we don't actually call `#remove_demo_data` and `#ensure_initial_objects`.

So then the last option would be to get rid of the `allow_any_instance_of` and also stub the `#worker` method ... That makes the test that asserts that we change the zone of the provider fail because somehow we are getting this "default" zone in there.